checkov/common/checks_infra/checks_parser.py

```diff
@@ -229,8 +229,16 @@ def _parse_raw_check(self, raw_check: Dict[str, Any], resources_types: Optional[
                     or (isinstance(resource_type, list) and resource_type[0].lower() == "all")
             ):
                 check.resource_types = resources_types or []
-            elif "provider" in resource_type and provider:
-                check.resource_types.append(f"provider.{provider}")
+
+            elif "provider" in resource_type:
+                provider = json_check.get("scope", {}).get("provider", "")
+                provider_type = ""
+                if cast(str, provider):
+                    provider_type = provider.lower()
+                elif cast(list, provider):
+                    provider_type = provider[0].lower()
+                check.resource_types.append(f"provider.{provider_type}")
```

Comment:
@reemodo The provider handling should occur on line 188, not downstream. If we support only 1 provider at a time, and it could be a list or a string that doesn't matter to the rest of the parser at all